appservice/src/deploy/getDeployNode.ts

```diff
@@ -0,0 +1,53 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { AzExtTreeItem } from 'vscode-azureextensionui';
+import { ext } from '../extensionVariables';
+import { localize } from '../localize';
+import { getWorkspaceSetting, updateWorkspaceSetting } from '../utils/settings';
+import { AppSource, IDeployContext } from './IDeployContext';
+
+export async function getDeployNode<T extends AzExtTreeItem>(context: IDeployContext, arg1: unknown, arg2: unknown, expectedContextValue: string | string[]): Promise<T> {
+    let node: AzExtTreeItem | undefined;
+
+    if (arg1 instanceof AzExtTreeItem) {
+        node = arg1;
+        context.appSource = AppSource.tree;
+    } else if (typeof arg2 === 'string' && arg2) {
```

Comment:
Do we need to verify arg2 is defined if it is typeof string?

Reply:
Uh this was more for the 'empty string' case than undefined